core-plugins/src/test/java/io/cdap/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -982,6 +1001,41 @@ public void testFileBatchInputFormatParquetMissingField() throws Exception {
     Assert.assertEquals(expected, output);
   }
 
+  @Test
+  public void testDelimitedFormatsValidSchema() throws Exception {
+    FileSourceConfig config = new FileSourceConfig();
+    FailureCollector collector = new MockFailureCollector();
+
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("referenceName"), "ref");
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("format"), "csv");
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("schema"),
+                         RECORD_SCHEMA_VALID_DATE.toString());
+    FieldSetter.setField(config, FileSourceConfig.class.getDeclaredField("path"), "path");
+
+    config.validate(collector);
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testDelimitedFormatsInValidSchema() throws Exception {
+    FileSourceConfig config = new FileSourceConfig();
+    FailureCollector collector = new MockFailureCollector();
+
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("referenceName"), "ref");
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("format"), "csv");
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("schema"),
+                         RECORD_SCHEMA_INVALID_DATE.toString());
+    FieldSetter.setField(config, FileSourceConfig.class.getDeclaredField("path"), "path");
+
+    config.validate(collector);
+    Assert.assertEquals(1, collector.getValidationFailures().size());
+
+    List<ValidationFailure.Cause> failureCauses = collector.getValidationFailures().get(0).getCauses();
+    Assert.assertEquals(2, failureCauses.size());
+    Assert.assertEquals("format", failureCauses.get(0).getAttribute("stageConfig"));
+    Assert.assertEquals("date", failureCauses.get(1).getAttribute("outputField"));
```

Comment:
We can also catch the error message in here.

Reply:
PR updated.